elasticapm/contrib/serverless/aws.py

```diff
@@ -50,11 +51,65 @@
 logger = get_logger("elasticapm.serverless")
 
 COLD_START = True
+INSTRUMENTED = False
 
-_AnnotatedFunctionT = TypeVar("_AnnotatedFunctionT")
+_AWSLambdaContextT = TypeVar("_AWSLambdaContextT")
 
 
-class capture_serverless(object):
+def capture_serverless(func: Optional[callable] = None, **kwargs) -> callable:
+    if not func:
+        return functools.partial(capture_serverless, **kwargs)
+
+    if kwargs:
+        warnings.warn(
+            PendingDeprecationWarning(
+                "Passing keyword arguments to capture_serverless will be deprecated in the next major release."
+            )
+        )
+
+    name = kwargs.pop("name", None)
+
+    # Disable all background threads except for transport
+    kwargs["metrics_interval"] = "0ms"
+    kwargs["breakdown_metrics"] = False
+    if "metrics_sets" not in kwargs and "ELASTIC_APM_METRICS_SETS" not in os.environ:
+        # Allow users to override metrics sets
+        kwargs["metrics_sets"] = []
+    kwargs["central_config"] = False
+    kwargs["cloud_provider"] = "none"
+    kwargs["framework_name"] = "AWS Lambda"
+    if "service_name" not in kwargs and "ELASTIC_APM_SERVICE_NAME" not in os.environ:
+        kwargs["service_name"] = os.environ["AWS_LAMBDA_FUNCTION_NAME"]
+    if "service_version" not in kwargs and "ELASTIC_APM_SERVICE_VERSION" not in os.environ:
+        kwargs["service_version"] = os.environ.get("AWS_LAMBDA_FUNCTION_VERSION")
+
+    global INSTRUMENTED
+    client = get_client()
+    if not client:
+        client = Client(**kwargs)
+    if not client.config.debug and client.config.instrument and client.config.enabled and not INSTRUMENTED:
+        elasticapm.instrument()
+        INSTRUMENTED = True
```

Comment:
Technically this `INSTRUMENTED` flag is not required. Instrumentation is idempotent, so multiple calls don't matter. I added it purely for performance reasons.